### PR TITLE
Resource quotas frontend implementation

### PR DESCRIPTION
**Pull request by batikanu**
- Resource quota frontend implemented. 
- Resource quota backend refactored and missing scopes added. 
- Resource quota list added to namespace detail page

### CODE REVIEW

**Comment by bryk** [comment]
Please don't yet merge. When you're done reviewing, let me know. I'll merge manually.

**Comment by floreks** [comment]
'Resource quotas' section is missing on admin page. Please add.

**Comment by batikanu** [comment]
Admin section added. PTAL

**Comment by batikanu** [comment]
rqexample

**Comment by batikanu** [comment]
The only problem is we don't have such a component this is a mix of info card and content card.

**Comment by batikanu** [comment]
I've updated the code with the design above. Shall I remove resource quota listing backend apis? They are not used at the moment but may be used later.

**Comment by bryk** [comment]
@batikanu Sorry for late review here. I was on vacation. 

@rf232 Can you help me reviewing this?

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<div layout="column">
+    <kd-content-card>
+        <kd-title>{{::$ctrl.i18n.MSG_RESOURCE_QUOTA_INFO_STATUS_SECTION}}</kd-title>
```
Comment (bryk) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: Can you use the new i18n framework?

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            <div layout="column" class="layout-column">
+                <div class="kd-info-card-section layout-padding layout-row"
+                     layout="row" layout-padding="">
+                    <div layout="column"
+                         class="kd-info-card-section-container layout-column">
+                        <div flex="auto" class="flex-auto">
+                            <div layout="row"
+                                 class="kd-info-card-entry layout-row">
+                                <div flex="nogrow" class="ng-binding ng-scope flex-nogrow">
+                                    Name:
```
Comment (bryk) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: Can you i18n this?

**Comment by bryk** [comment]
> Shall I remove resource quota listing backend apis? They are not used at the moment but may be used later.

I'd say yes. If something is not used not it should not be in the repo :)

**Comment by batikanu** [comment]
> I'm looking at this and wonder whether this should be N boxes with resource quotas. Or one box called resource quotas with a list of it. I'd opt for the latter. But then we have a problem of how to design it. Can you propose something here?

If there is only one resource quota for one namespace as it has to be, it will show only one card. It will only show more cards if the user creates multiple resource quotas for the same namespace. It has the same behavior with kubectl right now.  

The simplest solution might be merging all resources into one table with one card but we can't display resource quota names.

File: src/app/frontend/namespacedetail/namespacedetail.html [hunk 3]
```diff
@@ -15,6 +15,12 @@
 -->
 
 <kd-namespace-info namespace="::ctrl.namespaceDetail"></kd-namespace-info>
+<div ng-if="ctrl.namespaceDetail.resourceQuotaList.items">
+    <div ng-repeat="(key, value) in ::ctrl.namespaceDetail.resourceQuotaList.items">
```
Comment (bryk) [inline src/app/frontend/namespacedetail/namespacedetail.html]: No need for `(key, value) in ...` you can just `resourceQuota in ...`.

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html (hunk 2 above)
Comment (bryk) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: Done or am I looking at a wrong diff? I still see `Name:` only.

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                        </div>
+                    </div>
+                    <div layout="column"
+                         class="kd-info-card-section-container layout-column">
+                        <div flex="auto" class="flex-auto">
+                            <div layout="row"
+                                 class="kd-info-card-entry layout-row">
+                                <div flex="nogrow"
+                                     class="ng-binding flex-nogrow">
+                                    Scopes:
```
Comment (bryk) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: I18n this?

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                                </div>
+                                <div flex="auto" class="kd-info-card-entry-content flex-auto">
+                                    <kd-middle-ellipsis ng-repeat="(key, value) in ::$ctrl.resourceQuotaDetail.scopes"
```
Comment (bryk) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: Just `scope in ...` would work.

File: src/app/frontend/resourcequotadetail/resourcequotadetail.html (hunk 4 above)
Comment (batikanu) [inline src/app/frontend/resourcequotadetail/resourcequotadetail.html]: Sorry, apparently I missed hard coded ones :)

File: src/app/frontend/namespacedetail/namespacedetail.html (hunk 3 above)
Comment (batikanu) [inline src/app/frontend/namespacedetail/namespacedetail.html]: Done

**Comment by batikanu** [comment]
PTAL. I can do it editable in next PR. I will create an issue as a reminder after merging this.

File: src/app/frontend/namespacedetail/namespacedetail.html (hunk 3 above)
Comment (batikanu) [inline src/app/frontend/namespacedetail/namespacedetail.html]: Done

**Comment by bryk** [comment]
Great, thanks! I'll do final check and merge if all is good :)

**Pull request merged**